invoice/src/lib.rs

```diff
@@ -124,3 +126,46 @@ pub enum ParseInvoiceError {
     #[from]
     Bp(bp::ParseWitnessOutError),
 }
+
+pub struct RgbInvoice<T: Display + FromStr>(CallRequest<T, RgbBeneficiary>);
+
+impl<T: Display + FromStr> RgbInvoice<T> {
+    pub fn new(
+        scope: T,
+        beneficiary: RgbBeneficiary,
+        // Core parameters
+        value: Option<u64>,
+        expiry_time: Option<DateTime<Utc>>,
+        // Additional fields
+        api: Option<TypeName>,
+        call: Option<CallState>,
+        lock: Option<TinyBlob>,
+        endpoints: Option<impl Into<ConfinedVec<Endpoint, 0, 10>>>,
```

Comment:
this rather should be
```suggestion
        endpoints: impl IntoIterator<Endpoint>,
```

And the method should panic if too many endpoints are provided